pkg/controller/installation/core_controller.go

```diff
@@ -1783,3 +1797,57 @@ func removeInstallationFinalizer(i *operator.Installation) {
 		i.SetFinalizers(stringsutil.RemoveStringInSlice(CalicoFinalizer, i.GetFinalizers()))
 	}
 }
+
+func (r *ReconcileInstallation) updateInstallationStatus(instance *operator.Installation, conditions []operator.TigeraStatusCondition) {
+	if instance.Status.Conditions == nil {
+		instance.Status.Conditions = []metav1.Condition{}
+	}
+
+	for _, condition := range conditions {
+		found := false
+
+		ctype := string(condition.Type)
+		if condition.Type == operator.ComponentAvailable {
+			ctype = string(operator.ConditionTypeReady)
+		}
+		status := metav1.ConditionUnknown
+		if condition.Status == operator.ConditionTrue {
+			status = metav1.ConditionTrue
+		} else if condition.Status == operator.ConditionFalse {
+			status = metav1.ConditionFalse
+		}
+		ic := metav1.Condition{
+			Type:               ctype,
+			Status:             status,
+			LastTransitionTime: condition.LastTransitionTime,
+			ObservedGeneration: instance.GetGeneration(),
+		}
+
+		if len(condition.Reason) > 0 {
+			ic.Reason = condition.Reason
+		} else {
+			ic.Reason = string(operator.NotAvailable)
+		}
+		if len(condition.Message) > 0 {
+			ic.Message = condition.Message
+		} else {
+			ic.Message = "Not Available"
```

Comment:
Not Available might imply that the resource or deployment or whatever is not available.

If I am not mistaken message is not a mandatory field. We could keep the message empty or otherwise we could use "Unknown"

Reply:
Message is a required filed - https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#Condition